src/uniprotkb/components/entry/similar-proteins/SimilarProteins.tsx

```diff
@@ -83,56 +85,46 @@ const SimilarProteins: FC<{
-                              </h5>
-                              <ul className="no-bullet">
-                                {/* Note: move following to its own component  */}
-                                {row.members
-                                  ?.filter(
-                                    (member) => member !== representativeId
-                                  )
-                                  .map((member) => (
-                                    <li key={member}>
-                                      <Link
-                                        to={generatePath(
-                                          LocationToPath[
-                                            Location.UniProtKBEntry
-                                          ],
-                                          {
-                                            accession: member,
-                                          }
-                                        )}
-                                      >
-                                        {member}
-                                      </Link>
-                                    </li>
-                                  ))}
-                                {row.members &&
-                                  row.memberCount > row.members?.length && (
-                                    <li>{row.memberCount} more</li>
-                                  )}
-                              </ul>
-                            </section>
-                          )
+                          (row) => {
+                            const unirefEntryUrl = generatePath(
+                              LocationToPath[Location.UniRefEntry],
+                              {
+                                accession: row.id,
+                              }
+                            );
+                            return (
+                              <section key={row.id}>
+                                <h5>
+                                  <Link to={unirefEntryUrl}>{row.id}</Link>
+                                </h5>
+                                <SimilarProteinsTable members={row.members} />
+                                {row.memberCount - row.members.length > 0 && (
+                                  <Link to={unirefEntryUrl}>
+                                    {row.memberCount - row.members.length} More
+                                  </Link>
+                                )}
+                              </section>
+                            );
+                          }
                         )}
                         <hr />
                       </section>
                     )
                   )}
-                  {/* TODO: check with backend what query should be generated here */}
-                  <Button>View all</Button>
+                  {/* TODO: This query doesn't seem to work currently */}
+                  <Button
+                    onClick={() => {
```

Comment:
This needs to be a link.
```
                  <Button
                    component={Link}
                    to={{
                        pathname: LocationToPath[Location.UniProtKBResults],
                        search: `query=(${data?.results
                          .filter(({ entryType }) => entryType === clusterType)
                          .map(({ id }) => `uniref_cluster_100:${id}`)
                          .join(' OR ')})`,
                    }}
                  >
```

Reply:
👆🏽

Reply:
Ok I'm a little confused and am going to have another look at `<Button>`